9.2/run-postgresql.sh

```diff
@@ -1,12 +1,23 @@
 #!/bin/bash
 
+# Set HOME variable
+export HOME=/var/lib/pgsql
+
 # For SCL enablement
 source $HOME/.bashrc
 
 set -eu
 
 # Data dir
-export PGDATA=$HOME/data
+if [ -O $HOME/data ]
+then
+  export PGDATA=$HOME/data
+else
+  # If current user does not own data directory
+  # create a subdirectory that the user does own
+  mkdir $HOME/data/userdata
+  export PGDATA=$HOME/data/userdata
```

Comment:
In this case the user has write access to $HOME/data because it is now set as world-writeable by the Dockerfile. However, the user is not the owner. When postgres tries to start, it attempts to fix the permissions on its data directory and if the user is not the owner, it complains. The solution I found was to create a subdirectory under data that is owned by the current user.